src/main/java/seedu/address/model/tuitionclass/Time.java

```diff
@@ -10,14 +10,26 @@ public class Time {
     private final LocalTime startTime;
     private final LocalTime endTime;
 
+    public static final String MESSAGE_CONSTRAINTS =
+            "Starting time and ending time must be present in HH:MM and separated by a dash";
+    public static final String VALIDATION_REGEX = "\\p{Digit}{2}:\\p{Digit}{2}-\\p{Digit}{2}:\\p{Digit}{2}";
+    public final String timeFrame;
+
+
     /**
      * The constructor for a Time object.
      *
      * @param startTime A string representing the start time of the timeslot of a tuition class.
      * @param endTime A string representing the end time of the timeslot of a tuition class.
      */
     public Time(String startTime, String endTime) {
```

Comment:
May want to use try to detect exception when time passed is invalid like 99:99 which isValidTime could not detect. Also may want to check if endTime is earlier than startTime :)

Reply:
VALIDATION_REGEX was meant to just check if the input string matches the format, for whether the time is valid will depend on ParserUtil [line 130](https://github.com/AY2223S1-CS2103T-F12-4/tp/pull/53/files#r992669296)

Reply:
ok can. Time constructor will now catch DateTimeParseException within itself and throw a ParseException to ParseUtil::parseTime.
Also, it will also throw ParseException if startTime > endTime, i will allow startTime = endTime for now..
In future, I will want to improve the error message output and time format allowance.

Reply:
nice thanks!